server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -350,6 +381,87 @@ tableName, getPayloadType(), getCollation()
     );
   }
 
+  public void alterEntryTable(final String tableName)
+  {
+    try {
+      retryWithHandle(
+          new HandleCallback<Void>()
+          {
+            @Override
+            public Void withHandle(Handle handle)
+            {
+              if (!tableContainsColumn(handle, tableName, "type")) {
+                log.info("Adding column: type to table[%s]", tableName);
+                handle.execute(StringUtils.format("ALTER TABLE %1$s ADD COLUMN type VARCHAR(255)", tableName));
+              }
+              if (!tableContainsColumn(handle, tableName, "group_id")) {
+                log.info("Adding column: group_id to table[%s]", tableName);
+                handle.execute(StringUtils.format("ALTER TABLE %1$s ADD COLUMN group_id VARCHAR(255)", tableName));
+              }
+              return null;
+            }
+          }
+      );
+    }
+    catch (Exception e) {
+      log.warn(e, "Exception altering table");
+    }
+  }
+
+  @Override
+  public boolean migrateTaskTable()
+  {
+    final MetadataStorageTablesConfig tablesConfig = tablesConfigSupplier.get();
+    final String entryType = tablesConfig.getTaskEntryType();
+    final String tableName = tablesConfig.getEntryTable(entryType);
+    return migrateTaskTable(tableName);
+  }
+
+  public boolean migrateTaskTable(String tableName)
+  {
+    log.info("Populate fields task and group_id of task entry table [%s] from payload", tableName);
+    try {
+      retryWithHandle(
+          new HandleCallback<Void>()
+          {
+            @Override
+            public Void withHandle(Handle handle) throws SQLException, IOException
+            {
+              ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
`new ObjectMapper()` should never exist in the Druid code.  It's almost always incorrect and if you find yourself needing to do it because you have no other way of getting an ObjectMapper, the code structure is likely to blame.

In this case, the fact that the `MetadataConnector` is dealing with these things is to blame.  `SQLMetadataStorageActionHandler` already has an object mapper that it uses.  The logic for this migration should actually exist inside of that class or `MetadataTaskStorage` and *not* in here.

Reply:
done